Automattic/jetpack
# Update to pnpm 7

#### Changes proposed in this Pull Request:

This updates the monorepo to use pnpm 7. Notable changes required for this include:

* Remove `--` from `pnpm run` calls.
* Set `NODE_PATH` in various scripts.
* Add `eslint` as a dep to eslint-config-target-es and js-tools, as pnpm no longer does that for us.
* Clean up some pnpmfile entries, as pnpm 7.0.1 includes some of those fixes upstream.
* Add `require.resolve` where missing in some test configurations.
* Add new pnpmfile entries for some packages where missing peer deps are showing up now due to lack of `NODE_PATH` being set.

#### Jetpack product discussion


pdWQjU-13-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* Make sure various scripts still work.
* Build output should generally remain unchanged.
